model-optimizer/mo/utils/versions_checker.py

```diff
@@ -102,7 +143,7 @@ def get_module_version_list_from_file(file_name):
     req_dict = list()
     with open(file_name) as f:
         for line in f:
-            req_dict = parse_versions_list(line, req_dict)
+            req_dict = parse_versions_list(line, req_dict, env_params)
```

Comment:
For me it looks more logical to filter requirements in a separate step. So the `parse_versions_list` should return tuples with 4 elements (4-th element is None if there is no marker or a marker (list of markers) to check for) and a separate function should filter out requirements which does not satisfy the condition.

Reply:
We can rename this function. Now it parse and filter lines of requirements file. The other function checks versions of dependency.
I think that it will be also clear.

Reply:
I have called this function as `parse_and_filter_versions_list`